### PR TITLE
BaSP-MR-113: redux on members.

### PR DESCRIPTION
**Pull request by Franro212**
Implementation of Redux in members component.
Implementation of spiner in form and index.

### CODE REVIEW

**Comment by LucianoStradiot** [comment]
[Implement Redux on the Members entity](https://trello.com/c/7HDOkTzq/113-implement-redux-on-the-members-entity)

File: src/Redux/Members/reducer.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        ...state,
+        isPending: true,
+        error: null
+      };
+    }
+    case actionCreator.DELETE_MEMBERS_SUCCESS: {
+      return {
+        ...state,
+        isPending: false,
+        data: state.data.filter((member) => member._id !== action.payload.id),
```
Comment (Tilajc) [inline src/Redux/Members/reducer.js]: the id without _ is undefined
```suggestion
        data: state.data.filter((member) => member._id !== action.payload._id),
```

File: src/Redux/Members/constants.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export const GET_MEMBERS_PENDING = 'GET_MEMBERS_PENDING';
+export const GET_MEMBERS_SUCCESS = 'GET_MEMBERS_SUCCESS';
+export const GET_MEMBERS_ERROR = 'GET_MEMBERS_ERROR';
+export const POST_MEMBERS_PENDING = 'POST_MEMBERS_PENDING';
+export const POST_MEMBERS_SUCCESS = 'POST_MEMBERS_SUCCESS';
+export const POST_MEMBERS_ERROR = 'POST_MEMBERS_ERROR';
+export const PUT_MEMBERS_PENDING = 'PUT_MEMBERS_ERROR';
```
Comment (Tilajc) [inline src/Redux/Members/constants.js]: there is an infinite spinner loop
```suggestion
export const PUT_MEMBERS_PENDING = 'PUT_MEMBERS_PENDING';
```

File: src/Components/Members/FormMembers/index.js
Comment (Tilajc) [inline src/Components/Members/FormMembers/index.js:1]: You need to validate the response and when the response is OK you should go back to the list, and when it's !OK should remain in the page

File: src/Redux/Members/reducer.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        isPending: true,
+        error: null
+      };
+    }
+    case actionCreator.PUT_MEMBERS_SUCCESS: {
+      return {
+        ...state,
+        isPending: false,
+        data: state.data.map((member) =>
+          member._id === action.payload.id ? action.payload : member
```
Comment (Tilajc) [inline src/Redux/Members/reducer.js]: action.payload.id is undefined
```suggestion
          member._id === action.id ? action.payload : member
```

**Review by Tilajc — CHANGES_REQUESTED** [review]
I left some comments, but it's almost done, Good job!

File: src/Redux/Members/reducer.js (hunk 1 above)
Comment (Franro212) [reply src/Redux/Members/reducer.js]: done

File: src/Redux/Members/constants.js (hunk 2 above)
Comment (Franro212) [reply src/Redux/Members/constants.js]: donde

File: src/Components/Members/FormMembers/index.js
Comment (Franro212) [reply src/Components/Members/FormMembers/index.js:1]: done

File: src/Redux/Members/reducer.js (hunk 3 above)
Comment (Franro212) [reply src/Redux/Members/reducer.js]: done

**Review by Tilajc — APPROVED** [review]
Well done!

**Comment by LautaroVercellini** [comment]
Works Good!

**Pull request merged**